storage/replica.go

```diff
@@ -100,6 +100,9 @@ const (
 
 	// DefaultLeaderLeaseDuration is the default duration of the leader lease.
 	DefaultLeaderLeaseDuration = time.Second
+
+	// applyQueueChanSize is the length of applyQueueChan
+	applyQueueChanSize = 1000
```

Comment:
Channels eagerly allocate their buffers. A channel with 1000 pointers is as expensive as a goroutine, and a `chan committedCmd` is much larger. If we're going to have a channel per replica, it must have a small buffer.

Alternately, maybe it would make sense to have a fixed pool of say 4 or 8 goroutines, and distribute commands to them based on a hash of the range ID. One of the reasons we have been running all commands on a single goroutine per store is the assumption that all commands would be competing for the same underlying disk resources. Running (up to) one goroutine per range may be slower than a limited number of goroutines.